invenio_rdm_records/requests/record_deletion.py

```diff
@@ -0,0 +1,148 @@
+
+
+class CreateAction(actions.CreateAction):
+    """Create action."""
+
+    def execute(self, identity, uow):
+        """Execute the create action."""
+        record = self.request.topic.resolve()
+        existing_requests = self._get_existing_requests(record)
+        if existing_requests.total > 0:
+            raise ValidationError(
+                _("A deletion request for this record already exists."),
+            )
+
+        self._verify_removal_reason()
+        self.request["title"] = _('Deletion request for "{record_title}"').format(
+            record_title=record.metadata["title"]
+        )
+
+        super().execute(identity, uow)
+
+    def _get_existing_requests(self, record):
+        """Return existing open deletion requests for the record."""
+        return current_requests_service.search(
+            system_identity,
+            extra_filter=dsl.Q(
+                "bool",
+                must=[
+                    dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                    dsl.Q("term", **{"type": RecordDeletion.type_id}),
+                    dsl.Q("term", **{"is_open": True}),
+                ],
+            ),
+        )
+
+    def _verify_removal_reason(self):
+        invalid_reason_msg = _("Invalid removal reason")
+        removal_reason_id = self.request.get("payload", {}).get("reason")
+        if not removal_reason_id:
+            raise ValidationError(_("Removal reason is required."))
+        try:
+            vocab = current_vocabularies_service.read(
+                identity=system_identity, id_=("removalreasons", removal_reason_id)
+            )
+            if "deletion-request" not in vocab.data.get("tags", []):
+                raise ValidationError(invalid_reason_msg)
+        except (PIDDoesNotExistError, NoResultFound):
+            raise ValidationError(invalid_reason_msg)
+
+
+class AcceptAction(actions.AcceptAction):
+    """Accept action."""
+
+    def execute(self, identity, uow, **kwargs):
+        """Delete the record."""
+        record = self.request.topic.resolve()
+        request_creator = self.request.created_by.resolve()
+
+        removal_reason_id = self.request.get("payload", {}).get("reason")
+        tombstone_data = {
+            "removal_reason": {"id": removal_reason_id},
+            "removed_by": {"user": str(request_creator.id)},
+        }
+
+        # For immediate deletions, we track in the tombstone the deletion policy ID
+        policy_id = self.request.get("payload", {}).get("policy_id")
+        if policy_id:
+            tombstone_data["note"] = str(_("Record deleted by uploader"))
+            # NOTE: Deletion policies are not a vocabulary, but we follow the same
+            # format for consistency and potential extensions.
+            tombstone_data["deletion_policy"] = {"id": policy_id}
+
+        current_rdm_records_service.delete_record(
+            identity, record["id"], tombstone_data, uow=uow
+        )
```

Comment:
We have to make sure that when an administrator accepts the request, `tombstone.removed_by` is not the user who created the request, but the administrator.